storage/rocksdb/ha_rocksdb.cc

```diff
@@ -5018,6 +5115,9 @@ int ha_rocksdb::open(const char *const name, int mode, uint test_if_locked) {
     DBUG_RETURN(HA_ERR_OUT_OF_MEM);
   }
 
+  m_ttl_bytes =
```

Comment:
hmmm, I know we talked about this offline but I think I misunderstood. `ttl_bytes` is the result of an `Rdb_string_reader.read` call, which returns pointer to the start of the bytes read.  The bytes read in this case are from `m_retrieved_record`.  However, in the code it is now possible for `m_retrieved_record` to change between the `convert_record_from_storage_format` (where `m_ttl_bytes` is populated) and `convert_record_to_storage_format` calls, this potentially happens on the write path.  This means if we store `ttl_bytes` into `m_ttl_bytes` directly (aka if they point to the same place), it can change from out under us, which we don't want.

So either way, we need to copy the content `ttl_bytes` somewhere.  Am I missing something?